systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1077,453 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+                .withTrustedCertificates(certSecretTarget)
+            .endTls()
+            .addToConfig("config.storage.replication.factor", -1)
+            .addToConfig("offset.storage.replication.factor", -1)
+            .addToConfig("status.storage.replication.factor", -1)
+            .build();
+
+        resourceManager.createResource(extensionContext, KafkaMirrorMaker2Templates.kafkaMirrorMaker2(testStorage.getClusterName(), kafkaClusterTargetName, kafkaClusterSourceName, 1, true)
+            .editSpec()
+                .withClusters(targetClusterWithScramSha512Auth, sourceClusterWithScramSha512Auth)
+                    .editFirstMirror()
+                        .editSourceConnector()
+                            .addToConfig("refresh.topics.interval.seconds", 1)
+                        .endSourceConnector()
+                    .endMirror()
+            .endSpec()
+            .build());
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(kafkaClientsPodName)
+            .withTopicName(topicSourceNameA)
+            .withNamespaceName(testStorage.getNamespaceName())
+            .withClusterName(kafkaClusterSourceName)
+            .withKafkaUsername(userSource.getMetadata().getName())
+            .withMessageCount(messagesCount)
+            .withListenerName(kafkaTlsScramListenerName)
+            .build();
+
+        int sent = internalKafkaClient.sendMessagesTls();
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicTargetNameA)
+            .withClusterName(kafkaClusterTargetName)
+            .withKafkaUsername(userTarget.getMetadata().getName())
+            .withConsumerGroupName(ClientUtils.generateRandomConsumerGroup())
+            .build();
+
+        LOGGER.info("Now messages should be mirrored to target topic and cluster");
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+        LOGGER.info("Messages successfully mirrored");
+
+        Map<String, String> mmSnapshot = DeploymentUtils.depSnapshot(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()));
+        LOGGER.info("Changing KafkaUser sha-password on KMM2 Source and make sure it rolled");
+        Secret passwordSource = new SecretBuilder()
+            .withNewMetadata()
+                .withName(kafkaUserSourceName)
+            .endMetadata()
+            .addToData("password", "c291cmNlLXBhc3N3b3Jk")
+            .build();
+        kubeClient().patchSecret(testStorage.getNamespaceName(), kafkaUserSourceName, passwordSource);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()), 1, mmSnapshot);
```

Comment:
```suggestion
        Map<String, String> mmSnapshot = DeploymentUtils.depSnapshot(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()));
        
        LOGGER.info("Changing KafkaUser sha-password on KMM2 Source and make sure it rolled");
        
        Secret passwordSource = new SecretBuilder()
            .withNewMetadata()
                .withName(kafkaUserSourceName)
            .endMetadata()
            .addToData("password", "c291cmNlLXBhc3N3b3Jk")
            .build();       
        kubeClient().patchSecret(testStorage.getNamespaceName(), kafkaUserSourceName, passwordSource);
 
        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()), 1, mmSnapshot);
```

just for better visibility